components/feature/tabs/src/main/java/mozilla/components/feature/tabs/tabstray/TabsFeature.kt

```diff
@@ -37,4 +40,8 @@ class TabsFeature(
         presenter.stop()
         interactor.stop()
     }
+
+    fun invalidate() {
```

Comment:
Naming: We're not "invalidating" the feature. We just want it to update the displayed sessions / invalidate the UI.

Testing: This doesn't seem to be covered by a test?

Design: I think providing a lambda to filter sessions is fine, but the fact that we need a public API call on the feature to make it re-render the UI seems a bit weird. It seems the presenter should be able to observe some state and decide that it needs to that instead of getting nudge from the outside? But maybe that's for a future refactoring. Just a thought.

Reply:
> Design: I think providing a lambda to filter sessions is fine, but the fact that we need a public API call on the feature to make it re-render the UI seems a bit weird. It seems the presenter should be able to observe some state and decide that it needs to that instead of getting nudge from the outside? But maybe that's for a future refactoring. Just a thought.

We talked about this offline and decided the best way to avoid having to expose and 'invalidate' is to pass in the filter to feature when we need it to be invoked. So instead of providing the filter in the constructor AND exposing an invalidation method, we just have an `filterTabs` method that takes a filter and invokes the invalidation after that.